### DISCUSSION TITLE
Prettier print for BeEquivalentTo output

### DISCUSSION THREAD

**Discussion by rcdailey** [fluentassertions/fluentassertions#1722] [Q&A] [answered]
When I have a `BeEquivalentTo()` comparison between two lists of strings, the output looks like this:

```txt
Expected DynamicMenuTestHelpers.FixupCaptions(menu.ItemLinks) to be a collection with 4 item(s), but {"Generate Report (DR Objects)", "Generate Report (Non-DR Objects)"}"
"contains 2 item(s) less than"
"{"Schedule Disaster Recovery Backups...", "Run Backup Now", "Generate Report (DR Objects)", "Generate Report (Non-DR Objects)"}.
```

I personally find this difficult to read, mainly because everything is on one line. Is there a way to make this print out "prettier"? Something as simple as one string per line I think might make this more legible.

**Comment by dennisdoomen** [comment, marked as answer]
I agree. Ideally it should be smart enough to see that some of these strings are pretty long and print them on separate lines. Internally we already have a threshold for a human readable string that is set to a max of 8 characters.

**Reply by rcdailey** [reply]
I originally posed this as a question just in case there's some user-configurable way (or workaround) to enhance the assertion output. But if not, I'm happy to move this to a github issue if you prefer. Just let me know. And thank you for the response!

**Comment by dennisdoomen** [comment]
Already did this. See https://github.com/fluentassertions/fluentassertions/issues/1726

**Comment by ronaldkroon** [comment]
What you can do already:
`using var scope = new AssertionScope().UsingLineBreaks;`

**Reply by rcdailey** [reply]
I tried this above my `BeEquivalentTo()` assertion but I didn't notice a difference. Object graphs with more than just strings get really challenging to read. I can't even make use of it and often times have to resort to debug inspection to see the differences.
![image](https://github.com/fluentassertions/fluentassertions/assets/1768054/45ffa796-e22f-4ed2-b99d-83d84192f5d2)

**Reply by dennisdoomen** [reply]
We haven't done anything on this topic yet. See https://github.com/fluentassertions/fluentassertions/issues/1726

**Reply by rcdailey** [reply]
I'm sorry for the misunderstanding. I understand that nothing has changed in FluentAssertions yet for this. I was replying to @ronaldkroon's suggestion to use `UsingLineBreaks` which I expected to change the output in some way, but I didn't notice a difference. Good to hear from you Dennis and thank you for the reply!
